src/app/server/CommissioningWindowManager.cpp

```diff
@@ -0,0 +1,276 @@
+ */
+
+#include <app/server/CommissioningWindowManager.h>
+#include <app/server/Mdns.h>
+#include <app/server/Server.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <transport/SecureSessionMgr.h>
+
+#if CHIP_ENABLE_OPENTHREAD
+#include <platform/ThreadStackManager.h>
+#endif
+#include <lib/mdns/Advertiser.h>
+
+namespace {
+
+void HandleCommissioningWindowTimeout(chip::System::Layer * aSystemLayer, void * aAppState)
+{
+    chip::CommissioningWindowManager * commissionMgr = static_cast<chip::CommissioningWindowManager *>(aAppState);
+    commissionMgr->CloseCommissioningWindow();
+}
+
+void OnPlatformEventWrapper(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    chip::CommissioningWindowManager * commissionMgr = reinterpret_cast<chip::CommissioningWindowManager *>(arg);
+    commissionMgr->OnPlatformEvent(event);
+}
+} // namespace
+
+namespace chip {
+
+void CommissioningWindowManager::OnPlatformEvent(const DeviceLayer::ChipDeviceEvent * event)
+{
+    if (event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete)
+    {
+        if (event->CommissioningComplete.status == CHIP_NO_ERROR)
+        {
+            ChipLogProgress(Discovery, "Commissioning completed successfully");
+        }
+        else
+        {
+            ChipLogError(Discovery, "Commissioning errored out with error %" CHIP_ERROR_FORMAT,
+                         event->CommissioningComplete.status.Format());
+        }
+        // reset all advertising
+        app::MdnsServer::Instance().StartServer(Mdns::CommissioningMode::kDisabled);
+    }
+    else if (event->Type == DeviceLayer::DeviceEventType::kOperationalNetworkEnabled)
+    {
+        app::MdnsServer::Instance().AdvertiseOperational();
+        ChipLogError(Discovery, "Operational advertising enabled");
+    }
+}
+
+void CommissioningWindowManager::Cleanup()
+{
+    mServer->GetExchangManager().UnregisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::PBKDFParamRequest);
+    StopAdvertisement();
+
+    // reset all advertising
+    app::MdnsServer::Instance().StartServer(Mdns::CommissioningMode::kDisabled);
+}
+
+void CommissioningWindowManager::OnSessionEstablishmentError(CHIP_ERROR err)
+{
+    Cleanup();
+
+    ChipLogProgress(AppServer, "OnSessionEstablishmentError: %s", ErrorStr(err));
+    ChipLogProgress(AppServer, "Failed in SPAKE2+ handshake");
+
+    if (mAppDelegate != nullptr)
+    {
+        mAppDelegate->OnRendezvousStopped();
+    }
+}
+
+void CommissioningWindowManager::OnSessionEstablished()
+{
+    CHIP_ERROR err = mServer->GetSecureSessionManager().NewPairing(
+        Optional<Transport::PeerAddress>::Value(mPairingSession.GetPeerAddress()), mPairingSession.GetPeerNodeId(),
+        &mPairingSession, SecureSession::SessionRole::kResponder, 0);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Ble, "Failed in setting up secure channel: err %s", ErrorStr(err));
```

Comment:
is this only Ble or could it be IP as well (in which case AppServer maybe is more consistent as a log source)?